php/src/Google/Protobuf/Internal/Message.php

```diff
@@ -705,6 +744,62 @@ private function convertJsonValueToProtoValue(
                     $submsg->mergeFromJsonArray($value);
                 }
                 return $submsg;
+            case GPBType::BYTES:
+                if (!is_string($value)) {
+                    throw new GPBDecodeException("Expect string");
+                }
+                $proto_value = base64_decode($value, true);
+                if ($proto_value === false) {
+                    throw new GPBDecodeException(
+                        "Invalid base64 characters");
+                }
+                return $proto_value;
+            case GPBType::BOOL:
+                if ($is_map_key) {
+                    if ($value === "true") {
+                        return true;
+                    }
+                    if ($value === "false") {
+                        return false;
+                    }
+                    throw new GPBDecodeException(
+                        "Bool field only accept bool value");
+                }
+                if (!is_bool($value)) {
+                    throw new GPBDecodeException(
+                        "Bool field only accept bool value");
+                }
+                return $value;
+        }
+        return $this->convertNativeValueToProtoValue(
+            $value,
+            $field,
+            $is_map_key);
+    }
+
+    private function convertNativeValueToProtoValue(
+        $value,
+        $field,
+        $is_map_key = false)
+    {
+        if (is_null($value)) {
+            return $this->defaultValue($field);
+        }
+        switch ($field->getType()) {
+            case GPBType::MESSAGE:
+                $klass = $field->getMessageType()->getClass();
+                if ($value instanceof $klass) {
+                    return $value;
+                }
+                if (!is_object($value) && !is_array($value)) {
```

Comment:
If $value is array, we also need to handle that.

Reply:
Done. We handle this below when we call `mergeFromArray`.
Also I updated error message to "Expect message or array."